### DISCUSSION TITLE
CGNS development version won't install

### DISCUSSION THREAD

**Discussion by SRkumar97** [CGNS/CGNS#350] [CGNStalk]
Hello,
I need CGNS to be installed fully and in parallel.
I presume this site (https://cgns.github.io/download.html) is the one to be followed for download and installation. I am new to CGNS and haven't used before, so sorry for any mistake! My system is a single boot - Ubuntu 20.04 system
Here, I chose to use the development version of the code.
 git clone https://github.com/CGNS/CGNS.git
Now, the README file in the CGNS folder instructs to use **cmake /path to src folder/** to begin the compilation. However once i change working directory to /src folder and follow with this cmake command, there are errors related to the compilation script file and on a variable related to system configuration (x64). 
Have already installed HDF% in parallel, in the same directory as CGNS folder.

Requesting support and guidance with a clean and proper installation of CGNS! Thanks
![CGNS-development-cmake error](https://user-images.githubusercontent.com/109019272/178204861-219b6e51-b44e-430d-be15-b4a08d558869.png)

**Comment by brtnfld** [comment]
CMake should point to the top-level directory (i.e., the directory where you cloned CGNS into), not the "src" directory. I guess /path/to/cgns/sources might be confusing.

**Reply by SRkumar97** [reply]
Hello, thanks for the help. What I did was change directory one level up, to CGNS itself and then ran the command

cmake /home/labsys-ram/Hifiles/CGNS/

Now most of it looked good, just that error message is popping HDF5 library not found/missing. 
May I know where is the HDF5 path to be specified before the cmake command? Also are we trying to access any specific executable/header within HDF5?
Thanks

**Comment by brtnfld** [comment]
Did you set the install directory when building HDF5? Otherwise, it would depend on if you used cmake or Autotools to install HDF5 if you did not install it.
If you use ccmake, the path to the HDF5 install directory can be set there. You can also set the env variable "HDF5_ROOT" to the path of the HDF5 install directory so that CMake can find it. The HDF5 install dir. usually has "include" and "lib" directories.

**Comment by SRkumar97** [comment]
Hello, 
Thanks for the help. HDF5 was installed using cmake.
Now, the point I'm trying to convey is that, despite having set HDF5 as an environment variable in the bashrc file, CGNS won't configure.
The way environment variables for HDF5 have been set are -

#HDF5
export HDF5_ROOT=$HDF5_ROOT:/home/labsys-ram/Hifiles/hdf5-1.12.2/hdf5/
export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/home/labsys-ram/Hifiles/hdf5-1.12.2/hdf5/lib

I did only this much; did not modify/type path in any of the files and folders within CGNS
The attached screenshot shows the error I'm facing with the command
**cmake /home/labsys-ram/Hifiles/CGNS/**

Requesting your help.
![new_CGNS](https://user-images.githubusercontent.com/109019272/178423164-eff1e37a-f69b-4afe-b502-53acce55b19a.png)

**Comment by brtnfld** [comment]
I'm not sure. Here is my script for building a parallel version of CGNS. You can try modifying it. What version of CMake are you using? Can you list the contents of the HDF5 lib directory?
You can try setting HDF5_DIR instead. You can also try adding the -D CMAKE_PREFIX_PATH option.

```
#!/bin/bash
#
# This script uses cmake to build cgns, parallel build,
# it should be executed some place other then in the
# source directory.

#  Note:
#  Does szip support
HDF5=$HOME/packages/hdf5/build/hdf5
CGNS=../

export HDF5_DIR="$HDF5"
#export HDF5_ROOT="$HDF5"

cmake \
    -D CMAKE_C_COMPILER:PATH="mpicc" \
    -D CMAKE_C_FLAGS:STRING="-g" \
    -D CMAKE_Fortran_FLAGS:STRING="-g" \
    -D CMAKE_BUILD_TYPE:STRING="Debug" \
    -D CMAKE_Fortran_COMPILER:PATH="mpif90" \
    -D CGNS_BUILD_SHARED:BOOL=ON \
    -D CGNS_USE_SHARED:BOOL=ON \
    -D CMAKE_STATIC_LINKER_FLAGS:STRING="" \
    -D CGNS_ENABLE_TESTS:BOOL=ON \
    -D CGNS_BUILD_CGNSTOOLS:BOOL=OFF \
    -D CGNS_ENABLE_SCOPING:BOOL=OFF \
    -D CMAKE_INSTALL_PREFIX:PATH="$PWD/CGNS" \
    -D CMAKE_EXE_LINKER_FLAGS:STRING="" \
    -D CGNS_ENABLE_PARALLEL:BOOL=ON \
    -D HDF5_NEED_MPI:BOOL=ON \
    -D CGNS_ENABLE_LFS:BOOL=OFF \
    -D CGNS_ENABLE_HDF5:BOOL=ON \
    -D CGNS_ENABLE_FORTRAN:BOOL=ON \
    -D CGNS_ENABLE_64BIT:BOOL=ON \
    $CGNS

    # -D CMAKE_PREFIX_PATH=$HDF5 \
```

**Reply by SRkumar97** [reply]
Thanks, the version of CMake is 3.16.3
1) The contents within the parent HDF5 lib directory are shown in the pic -
![image](https://user-images.githubusercontent.com/109019272/178515191-cbf44178-61a1-418e-9487-b2f0daa750cd.png)

2) Now, when I type **ccmake .** in the CGNS root directory in terminal, the paths for several sets of environment variables/parameters are to be entered as shown in the rest 2 snips. Could you please check on the entries made for HDF5 variables?

![image](https://user-images.githubusercontent.com/109019272/178533640-1993b729-efa3-4303-b92f-af0bb679867d.png)
![image](https://user-images.githubusercontent.com/109019272/178540531-96d64942-0be7-4920-a66d-00c1f76ecb6d.png)

3) After this, **cmake \path to current CGNS folder** was entered. The code seems to detect HDF5 and build, but there are a series of almost same warnings and there is a message on missing HDF5 despite the fact that bashrc has env variables defined. How do I check if the build has been done properly in parallel?

#HDF5
export HDF5_ROOT='export PATH=/home/labsys-ram/Hifiles/hdf5-1.12.2/hdf5'
export HDF5_LIBRARIES='export PATH=/home/labsys-ram/Hifiles/hdf5-1.12.2/hdf5/lib'
export HDF5_DIR='export PATH=/home/labsys-ram/Hifiles/hdf5-1.12.2/hdf5'
![image](https://user-images.githubusercontent.com/109019272/178535672-2fdfaa8e-5547-490c-beb2-1e7e63dd161c.png)

Note that the list of variables in the **ccmake .** look quite different from the list you have shown. Am I missing out on something? 
Thanks a ton

**Comment by brtnfld** [comment]
I'm not sure why the CMAKE_INSTALL_PREFIX points to the HDF5 install directory.

Were these typos? They are not correct.
export HDF5_ROOT='export PATH=/home/labsys-ram/Hifiles/hdf5-1.12.2/hdf5'
export HDF5_LIBRARIES='export PATH=/home/labsys-ram/Hifiles/hdf5-1.12.2/hdf5/lib'
export HDF5_DIR='export PATH=/home/labsys-ram/Hifiles/hdf5-1.12.2/hdf5'

CMake HDF5 find packages has two paths. First, if the HDF5 build was done with CMake, it should find it and report libs: 1 static and 1 shared. 
If HDF5 were built using Autotools, it would report HDF5 not found and then later found it.

Using ccmake after running by build script shows the path for HDF5_DIR.

What happens when you try to make CGNS?

**Comment by SRkumar97** [comment]
Thanks, actually the errors got rectified after I used the install script you have provided, by uncommenting the HDF5_ROOT as well and then configuring.
Finally it showed built and we did a make test as well. Everything looks proper now.

![image](https://user-images.githubusercontent.com/109019272/178661447-456bf1e5-97e8-4ad8-95bf-cfa7b97dc63c.png)
The final lines of make command output is

[100%] Linking C executable write_grid_str_parinzone
[100%] Built target write_grid_str_parinzone
Scanning dependencies of target read_grid_str_parinzone
[100%] Building C object src/Test_UserGuideCode/C_code_parallel/CMakeFiles/read_grid_str_parinzone.dir/read_grid_str_parinzone.c.o
[100%] Linking C executable read_grid_str_parinzone
[100%] Built target read_grid_str_parinzone
Scanning dependencies of target read_flowcentrind_str_paroverzone
[100%] Building C object src/Test_UserGuideCode/C_code_parallel/CMakeFiles/read_flowcentrind_str_paroverzone.dir/read_flowcentrind_str_paroverzone.c.o
[100%] Linking C executable read_flowcentrind_str_paroverzone
[100%] Built target read_flowcentrind_str_paroverzone

Thanks a ton.
